### PR TITLE
[Merged by Bors] - chore(Algebra/Lie): rename IsSemisimple to HasTrivialRadical

### PR DESCRIPTION
**Pull request by jcommelin**
This frees up the name `IsSemisimple` so that it can be used for the notion that matches the categorical concept.

In characteristic 0, this is equivalent to having a trivial radical.
In characteristic p > 0, there are counterexamples.

With this change, we depart from the terminology used by Seligman,
but we feel that the change is justified because we need a name for the categorical version of semisimplicity,
and the naming convention proposed in this PR is the "least surprising" one.

This PR also changes the definition of `LieModule.IsIrreducible` to exclude the trivial module.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/Semisimple.lean [hunk 1]
```diff
@@ -35,7 +38,7 @@ universe u v w w₁ w₂
 class LieModule.IsIrreducible (R : Type u) (L : Type v) (M : Type w) [CommRing R] [LieRing L]
     [LieAlgebra R L] [AddCommGroup M] [Module R M] [LieRingModule L M] [LieModule R L M] :
     Prop where
-  Irreducible : ∀ N : LieSubmodule R L M, N ≠ ⊥ → N = ⊤
+  irreducible : ∀ N : LieSubmodule R L M, N ≠ ⊥ → N = ⊤
```
Comment (ocfnash) [inline Mathlib/Algebra/Lie/Semisimple.lean:41]: Given that you're here, maybe this whole definition could be replaced with:
```lean
/-- A Lie module is irreducible if its only Lie submodules are `⊥` and `⊤` (which may coincide). -/
abbrev LieModule.IsIrreducible (R L M : Type*) [CommRing R] [LieRing L] [LieAlgebra R L]
  [AddCommGroup M] [Module R M] [LieRingModule L M] [LieModule R L M] : Prop :=
  IsSimpleOrder (LieSubmodule R L M)
```

File: Mathlib/Algebra/Lie/Semisimple.lean [hunk 2]
```diff
@@ -49,56 +52,82 @@ class IsSimple extends LieModule.IsIrreducible R L L : Prop where
...
-for general coefficients. We are following [Seligman, page 15](seligman1967) and using the label
-for the weakest of the various properties which are all equivalent over a field of characteristic
-zero. -/
+for general coefficients.
+
+For example [Seligman, page 15](seligman1967) uses the label for `LieAlgebra.HasTrivialRadical`,
+whereas we reserve it for Lie algebras that are a direct sum of simple Lie algebras.
+-/
+class HasTrivialRadical : Prop where
+  trivial_radical : radical R L = ⊥
```
Comment (ocfnash) [inline Mathlib/Algebra/Lie/Semisimple.lean:70]:
```suggestion
  radical_eq_bot : radical R L = ⊥
```

File: Mathlib/Algebra/Lie/Semisimple.lean [hunk 3]
```diff
@@ -49,56 +52,82 @@ class IsSimple extends LieModule.IsIrreducible R L L : Prop where
...
+[upon](https://mathoverflow.net/questions/149391/on-radicals-of-a-lie-algebra#comment383669_149391)
+for general coefficients.
+
+For example [Seligman, page 15](seligman1967) uses the label for `LieAlgebra.HasTrivialRadical`,
+the weakest of the various properties which are all equivalent over a field of characteristic zero.
+-/
 class IsSemisimple : Prop where
-  semisimple : radical R L = ⊥
-#align lie_algebra.is_semisimple LieAlgebra.IsSemisimple
+  spanning : sSup {I : LieIdeal R L | IsAtom I} = ⊤
```
Comment (ocfnash) [inline Mathlib/Algebra/Lie/Semisimple.lean:84]: According to [rule 7](https://leanprover-community.github.io/contribute/naming.html#general-conventions) the usual conventions apply to constructor fields too:
```suggestion
  sSup_isAtom_eq_top : sSup {I : LieIdeal R L | IsAtom I} = ⊤
```
(and similarly the other two fields)

File: Mathlib/Algebra/Lie/Semisimple.lean [hunk 4]
```diff
@@ -49,56 +52,82 @@ class IsSimple extends LieModule.IsIrreducible R L L : Prop where
...
-theorem isSemisimple_iff_no_solvable_ideals :
-    IsSemisimple R L ↔ ∀ I : LieIdeal R L, IsSolvable R I → I = ⊥ :=
-  ⟨fun h => sSup_eq_bot.mp h.semisimple, fun h => ⟨sSup_eq_bot.mpr h⟩⟩
-#align lie_algebra.is_semisimple_iff_no_solvable_ideals LieAlgebra.isSemisimple_iff_no_solvable_ideals
+-- TODO: show that the atomic ideals of a semisimple Lie algebra are simple
 
-theorem isSemisimple_iff_no_abelian_ideals :
-    IsSemisimple R L ↔ ∀ I : LieIdeal R L, IsLieAbelian I → I = ⊥ := by
-  rw [isSemisimple_iff_no_solvable_ideals]
+theorem hasTrivialRadical_iff_no_solvable_ideals :
```
Comment (ocfnash) [inline Mathlib/Algebra/Lie/Semisimple.lean:90]: Optional: this should really be a pair of theorems:

-  the forward direction takes arguments `[HasTrivialRadical R L]` `I`,  `[IsLieAbelian I]` and proves `I = ⊥` (can probably be `simp`)
- the backward direction is an alternate constructor for `HasTrivialRadical`

Similarly for `hasTrivialRadical_iff_no_abelian_ideals` below

**Review by ocfnash — APPROVED** [review]
Thanks for doing this!

Please feel free to merge once you've thought about the suggestions (and appeased `lake shake`).

bors d+

File: Mathlib/Algebra/Lie/Semisimple.lean (hunk 3 above)
Comment (jcommelin) [reply Mathlib/Algebra/Lie/Semisimple.lean:84]: The last field was already fine, no?

**Pull request closed without merging**